services/agora/src/components/post/views/CommentSection.vue

```diff
@@ -246,6 +253,19 @@ watch(currentClusterTab, async () => {
   requestedCommentSlugId.value = "";
 });
 
+watch(
+  () => props.mode,
+  (newVal, oldVal) => {
+    if (newVal !== oldVal && newVal === "comment")
+      if (isMounted) {
+        sortAlgorithm.value = "discover";
+        currentClusterTab.value = "all";
+        updateInfiniteScrollingList("discover");
+        requestedCommentSlugId.value = "";
+      }
+  }
+);
+
```

Comment:
Slow.
TODO: improve this

Reply:
Note: in the later design, we should not share the list of comments at all between the comment and analysis tabs, so this should be unecessary.